src/CsharpClient/QuixStreams.Telemetry/Models/Telemetry/Parameters/Codecs/TimeseriesDataCustomJsonCodec.cs

```diff
@@ -0,0 +1,303 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using QuixStreams.Transport.Codec;
+using QuixStreams.Transport.Fw.Codecs;
+
+namespace QuixStreams.Telemetry.Models.Telemetry.Parameters.Codecs
+{
+    /// <summary>
+    /// TimeseriesData Json Codec implementation
+    /// </summary>
+    public class TimeseriesDataJsonCodec : Codec<TimeseriesDataRaw>
+    {
+        private static readonly DefaultJsonCodec<TimeseriesDataRaw> BaseCodec =
+            new DefaultJsonCodec<TimeseriesDataRaw>();
+
+        /// <inheritdoc />
+        public override CodecId Id => BaseCodec.Id; // this is only a serialization codec, still valid JSON
+
+        private Converter JsonConverter = new Converter(JsonSerializer.CreateDefault());
+
+        /// <inheritdoc />
+        public override TimeseriesDataRaw Deserialize(byte[] contentBytes)
+        {
+            using (var memoryStream = new MemoryStream(contentBytes))
+            {
+                using (var streamReader = new StreamReader(memoryStream))
+                {
+                    using (var jsonReader = new JsonTextReader(streamReader))
+                    {
+                        return JsonConverter.ReadJson(jsonReader);
+                    }
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public override TimeseriesDataRaw Deserialize(ArraySegment<byte> contentBytes)
+        {
+            using (var memoryStream = new MemoryStream(contentBytes.Array, contentBytes.Offset, contentBytes.Count))
+            {
+                using (var streamReader = new StreamReader(memoryStream))
+                {
+                    using (var jsonReader = new JsonTextReader(streamReader))
+                    {
+                        return JsonConverter.ReadJson(jsonReader);
+                    }
+                }
+            }
+        }
+
+
+        private void Serialize1(StreamWriter streamWriter, TimeseriesDataRaw obj)
+        {
+            using (var jsonWriter = new JsonTextWriter(streamWriter))
+            {
+                jsonWriter.CloseOutput = false;
+                JsonConverter.WriteJson(jsonWriter, obj);
+            }
+        }
+
+        private void Serialize2(StreamWriter streamWriter, TimeseriesDataRaw obj)
+        {
+            using (var jsonWriter = new JsonTextWriter(streamWriter))
+            {
+                jsonWriter.CloseOutput = false;
+                new JsonSerializer().Serialize(jsonWriter, obj);
+            }
+        }
```

Comment:
Serialize2 doesn't seem to be used, and Serialize1 could have a better name

Reply:
fixed!